### PR TITLE
Merge SIMDEV-InterfaceCreation into Dev

### PR DESCRIPTION
**Pull request by bhageneder**

### CODE REVIEW

**Review by Mitchell-Conrad-20 — CHANGES_REQUESTED** [review]
See comments

File: src/config/config.cfg
Comment (stoneoxmike) [inline src/config/config.cfg:1]: Does the ./ work?

**Review by stoneoxmike — APPROVED** [review]
See comment, otherwise looks like a great framework to start

File: src/interfaces/iDetectorClass.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            print("Deconstructing Detector Object")
+
+        #### Public Methods ####
+
+        # Transceiver Getter
+        def getTransceiver(self):
+            return self.__current_transceiver
+        
+        # robotList Getter
+        def getRobotLlist(self):
```
[the remark this answers is not in the document]
Comment (bhageneder) [reply src/interfaces/iDetectorClass.py:24]: Pulled it from Detector so they were miss-spelt in detector, fixed both instances.
What do you mean by not in the codebase?

File: src/interfaces/iDetectorClass.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+class iDetector:
+        # Constructor
+        def __init__(self):
+            self.__current_transceiver = -1
```
[the remark this answers is not in the document]
Comment (bhageneder) [reply src/interfaces/iDetectorClass.py:4]: Consistent with public methods for both virtual and actual architecture

File: src/sim/controller/classes/vDetectorClass.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+from interfaces.iDetectorClass import iDetector
+
+
+class simDetector(iDetector):
```
[the remark this answers is not in the document]
Comment (bhageneder) [reply src/sim/controller/classes/vDetectorClass.py:4]: Resolved

File: src/sim/controller/threads/vdiscovery.py
[the remark this answers is not in the document]
Comment (bhageneder) [reply src/sim/controller/threads/vdiscovery.py:1]: Place holder for the time being. Folder would not be tracked without file inside.

File: src/config/config.cfg
Comment (bhageneder) [reply src/config/config.cfg:1]: Error when running main currently "index out of bound" when accessing transceivers. I believe this was fixed in N-Robots but I don't have that change rn. Will work around it and let you know

File: src/interfaces/iDetectorClass.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mitchell-Conrad-20) [reply src/interfaces/iDetectorClass.py:24]: these methods are not part of detector

[the remark this answers is not in the document]
Comment (bhageneder) [reply src/interfaces/iDetectorClass.py:24]: Gotcha, I must have moved them on accident

File: src/config/config.cfg
Comment (stoneoxmike) [reply src/config/config.cfg:1]: Don't spend too much time bringing over changes that already exist. We can merge and test it later, just noting it for now

File: src/interfaces/iDetectorClass.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mitchell-Conrad-20) [reply src/interfaces/iDetectorClass.py:24]: nah I thought about it a bit more, they got removed in the PR that went through _yesterday_, so you didnt do anything wrong

File: src/classes/DetectorClass.py [hunk 4]
```diff
@@ -7,15 +7,16 @@
...
 import os
 
-class Detector:
+
+class Detector():
         # Constructor
         # Parameters: Width of Output Frame, Height of Output Frame, Object Detection Model Name, List of Camera Names [e.g., [0, 1, ...]), render (default false), debug (default false)
-        def __init__(self, width, height, modelName, cameras, render = False, tracking=False, debug = False):
+        def __init__(self, width, height, cameras, render = False, tracking=False, debug = False):
+                super().init()
```
Comment (Mitchell-Conrad-20) [inline src/classes/DetectorClass.py]: does this still need super.init then?

**Review by Mitchell-Conrad-20 — APPROVED** [review]
As long as code still is fully functional im cool with the merge

File: src/classes/RobotClass.py [hunk 5]
```diff
@@ -1,4 +1,4 @@
-class Robot:
+class Robot():
```
Comment (Mitchell-Conrad-20) [inline src/classes/RobotClass.py:1]: remove () after Robot

File: src/config/config.cfg
Comment (bhageneder) [reply src/config/config.cfg:1]: Wasn't able to test, will just take DEV config for now and modify later

File: src/classes/RobotClass.py (hunk 5 above)
Comment (bhageneder) [reply src/classes/RobotClass.py]: resolved

**Pull request merged**